types/rpc-handler.ts

```diff
@@ -55,6 +62,27 @@ export class RPCHandler implements HandlerInterface {
     this.testRpcPerformance.bind(this);
   }
 
+  /**
+   * Loops through all RPCs for a given network id and returns a provider with the first successful network.
+   */
+  public async getFirstAvailableRpcProvider() {
+    const rpcList = [...networkRpcs[this._networkId].rpcs];
+    shuffleArray(rpcList);
```

Comment:
What's the purpose of shuffling the array?

Reply:
Simply to avoid always using the same providers in the same order, so failure is mitigated (it's always the same list in the same order otherwise). Not really needed, just thought it would be nice if for example the beginning of the list has unreliable providers.